### ISSUE TITLE
More automated way to generate match-up Data

### ISSUE THREAD

**Issue by NomadiCxac** [pvpoke/pvpoke#261]
Hi I am a new coder (learning from the Odin Project) working on a react flash card project (that is part of the curriculum) and wanted to know if there was an easy(ier) way to access/generate data for the top "x" best/counter matches for all possible Pokémon that are eligible for each cup. 

Example of intended goal: I input Medicham into my app search bar and then it will make a request to the [ Pokemon API ](https://pokeapi.co/) for cards of the top 10 best/worst matchups for the current meta / in general.

I have been looking to see if there was an API for this and if not then this will come as a feature request!

Otherwise the solutions I have looked into:

(Tried)
- Try to parse/decouple the Battle class from the overarching GameMaster object
- Try to find where to plug in a list of pokemon from the gameMaster.json to loop through and code logic to generate json data for each pokemon. (Super lost as I have never worked with php or a singleton design of this size and scope)

(Haven't Tried): 
- Use a webscraper

Would love any help or direction as I am looking to improve both as a PoGo battler and a developer. Any suggestions, feedback or criticisms are welcome! 

TL;DR: 

How do I get the json/csv data of all possible pokemon matchups without manually doing this for all pokemon on the site? 

Steps and desired result below:
![image](https://github.com/pvpoke/pvpoke/assets/38193946/fba46869-efd8-44d7-92d5-72173585c961)
![image](https://github.com/pvpoke/pvpoke/assets/38193946/d43ada49-a60b-463d-baf2-d84deb9a0858)

**Comment by pvpoke** [comment]
Hey, thanks for posting! Unfortunately there isn't an API available to obtain the data you're looking for at the moment. If you limit your search to Pokemon in the core meta or even top 100, you could plug those Pokemon into the Matrix Battle tool and store the resulting CSV file somewhere to look up your results.

https://pvpoke.com/battle/matrix/

Thanks for the feature request and good luck with your project!

**Comment by NomadiCxac** [comment]
Thanks for the reply!

I actually studied the code repo for a long while and figured something out with another dev. I had to really understand the Interface, gm, PokeSelectors, Battle and Pokemon classes. 

I managed to extract the data I need and just brute forced downloaded all the possible data (100 MB) using a sliced version of the gameMaster.json file for all possible Pokemon and then hijacked pokeselectors in the generateMultiBattleResults function by looping through all possible pokemon and then downloading and saving all possible match ups into a json on my local computer.

Thanks again for making this code open source. I could have copped out and just made something simple without any real utllity (for me haha). The battle sim is awesome and really helps get the end result I want.


Cheers!
